core/src/main/scala/com/avast/dapper/CassandraDao.scala

```diff
@@ -0,0 +1,35 @@
+package com.avast.dapper
+
+import java.time.{Duration, Instant}
+
+import com.datastax.driver.core.ConsistencyLevel
+
+import scala.concurrent.Future
+
+trait CassandraDao[PrimaryKey, Entity <: CassandraEntity[PrimaryKey]] {
+  def tableName: String
+
+  def get(primaryKey: PrimaryKey, queryOptions: ReadOptions = ReadOptions.Default): Future[Option[Entity]]
+
+  def save(instance: Entity, queryOptions: WriteOptions = WriteOptions.Default): Future[Unit]
+
+  def delete(instance: Entity, queryOptions: DeleteOptions = DeleteOptions.Default): Future[Unit]
+}
+
+case class ReadOptions(consistencyLevel: Option[ConsistencyLevel] = None)
```

Comment:
Shouldn't you move these classes somewhere else?

Reply:
Well.. I can but since they're basically used on this single place, I left it here.

Reply:
Right, I get that. But maybe if I was new to the project and I was looking for some type `ReadOptions` I wouldn't be looking inside `CassandraDao.scala` file - that was my point.